tools/fuzzer.cpp

```diff
@@ -0,0 +1,211 @@
+            << std::endl;
+  std::cout << " --json        create JSON."
+            << std::endl;
+  std::cout << " <iterations>  number of iterations. Default: 1"
+            << std::endl;
+  std::cout << " <threads>  number of threads. Default: 1"
+            << std::endl;
+}
+
+static inline bool isOption(char const* arg, char const* expected) {
+  return (strcmp(arg, expected) == 0);
+}
+
+static void addString(Builder& builder, RandomGenerator& randomGenerator) {
+  static auto &availableChars = "0123456789"
+                                "abcdefghijklmnopqrstuvwxyz"
+                                "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
+  size_t length = randomGenerator.mt64() % 1000;
+  std::string s;
+  s.reserve(length);
+  while (length--) {
+    s += availableChars[randomGenerator.mt64() % (sizeof(availableChars) - 1)];
+  }
+  builder.add(Value(s));
+}
+
+static void generateVelocypack(Builder& builder, size_t depth, RandomGenerator& randomGenerator) {
+  RandomBuilderAdditions randomBuilderAdds;
+  while (true) {
+    randomBuilderAdds = static_cast<RandomBuilderAdditions>(randomGenerator.mt64() % 8);
+    if (depth > 10 && randomBuilderAdds < 2) {
+      continue;
+    }
+    switch (randomBuilderAdds) {
+      case ADD_ARRAY: {
+        builder.openArray(randomGenerator.mt64() % 2 ? true : false);
+        size_t numMembers = randomGenerator.mt64() % 10;
+        for (size_t i = 0; i < numMembers; ++i) {
+          generateVelocypack(builder, depth + 1, randomGenerator);
+        }
+        builder.close();
+        break;
+      }
+      case ADD_OBJECT: {
+        builder.openObject(randomGenerator.mt64() % 2 ? true : false);
+        size_t numMembers = randomGenerator.mt64() % 10;
+        for (size_t i = 0; i < numMembers; ++i) {
+          std::string key = "test" + std::to_string(i);
+          builder.add(Value(key));
+          generateVelocypack(builder, depth + 1, randomGenerator);
+        }
+        builder.close();
+        break;
+      }
+      case ADD_BOOLEAN:
+        builder.add(Value(randomGenerator.mt64() % 2 ? true : false));
+        break;
+      case ADD_STRING:
+        addString(builder, randomGenerator);
+        break;
+      case ADD_NULL:
+        builder.add(Value(ValueType::Null));
+        break;
+      case ADD_UINT64:
+        builder.add(Value(randomGenerator.mt64()));
+        break;
+      case ADD_INT64: {
+        uint64_t uintValue = randomGenerator.mt64();
+        int64_t intValue;
+        memcpy(&intValue, &uintValue, sizeof(uintValue));
+        builder.add(Value(intValue));
+        break;
+      }
+      case ADD_DOUBLE: {
+        uint64_t uintValue = randomGenerator.mt64();
+        double doubleValue;
+        memcpy(&doubleValue, &uintValue, sizeof(uintValue));
+        builder.add(Value(doubleValue));
+      }
+      default:
+        break;
+    }
+    break;
+  }
+}
+
+int main(int argc, char* argv[]) {
+  VELOCYPACK_GLOBAL_EXCEPTION_TRY
+    std::vector<std::thread> threads;
```

Comment:
I suggest moving this line down to where `threads` is used first. There is no need to declare all variables at the beginning of a function.